### PR TITLE
Protein sequence download

### PR DESCRIPTION
**Pull request by imransl**
## Type

- [ ] Bug fix
- [x] New feature
- [ ] Improvement to existing feature

## Related JIRA Issue(s)
[ENSWBSITES-650](https://www.ebi.ac.uk/panda/jira/browse/ENSWBSITES-650)

## Description
This PR adds the protein sequence download to the protein list item. It duplicates most of the code of the `InstantDownloadTranscript`.

## Views affected
Entity viewer -> gene view -> protein view

### CODE REVIEW

**Comment by azangru** [comment]
I am confused. Shouldn't the Download component appear for every protein? It's currently not showing for proteins that do not have that bottom section (external references?)

image

The mockup includes examples of proteins without external references that still have the download component:

image

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
 
-.codingExonCount {
-  grid-column: 1/2;
+.proteinSummaryTopWrapper {
+  display: grid;
+  grid-template-columns: 300px 395px;
+}
+
+.downloadWrapper {
+  text-align: right;
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss:16]: Will the `display: flex; justify-content: flex-end` communicate the intention better?

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 
   > div {
-    margin: 10px 0 10px 0;
+    padding: 10px 0 10px 0;
   }
 }
 
-.codingExonCount {
-  grid-column: 1/2;
+.proteinSummaryTopWrapper {
```
Comment (azangru) [inline src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss:10]: just `proteinSummaryTop`?

Comment (imransl) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss]: ✔️

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss (hunk 1 above)
Comment (imransl) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss:16]: This didn't work unfortunately so it is till using CSS grid and text alignment

**Comment by imransl** [comment]
@azangru Not showing the download component when the protein summary data is not available was a mistake. I've fixed it and should show for all proteins now.

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss (hunk 1 above)
Comment (azangru) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss:16]: > This didn't work unfortunately

Works for me 🤷‍♂️ 

image

**Comment by azangru** [comment]
Shouldn't the download component in the protein without proteinSummary be also positioned flush right?

image

File: src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss (hunk 1 above)
Comment (imransl) [reply src/ensembl/src/content/app/entity-viewer/gene-view/components/proteins-list/proteins-list-item-info/ProteinsListItemInfo.scss:16]: It still doesn't properly in my case:

Screenshot 2020-07-07 at 12 03 52

The download component takes up the whole horizontal space as if it is a block element.

**Pull request merged**